deps/imgui/CMakeLists.txt

```diff
@@ -29,9 +29,11 @@ if("${POLYSCOPE_BACKEND_OPENGL3_GLFW}")
   target_include_directories(imgui PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/imgui/")
   target_include_directories(imgui PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/../glfw/include/")
   target_include_directories(imgui PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/../glad/include/")
-  
+
   target_link_libraries(imgui PRIVATE glfw)
-  
+
+  target_compile_definitions(imgui PUBLIC IMGUI_DISABLE_OBSOLETE_FUNCTIONS)
```

Comment:
Can we remove this? It's good to avoid obsolete functions in Polyscope's own code, but putting this in the cmake will cause downstream users's code to fail unnecessarily when they upgrade. They can always define this themselves if wanted.

Reply:
Sure. How about adding it globally when polyscope is a toplevel project? That way we can catch it in the CI, but don't impose it on client projects.